### ISSUE TITLE
Default numValues to 1 for optional options

### ISSUE THREAD

**Issue by eschwartz** [four43/admiral-cli#8]
Currently, if you want to make an option optional, you omit the last argument to `option()`:

```
cli.option('foo', 'What the foo?', '-f', '--foo', 'string');
```

In this case, here is the behavior I would expect:

```js
// actual arguments are: --foo bar
cli.option('foo', 'What the foo?', '-f', '--foo', 'string');
cli.parse();

cli.params.foo === 'bar';   // true  (expected)
```

But the actual behavior is this:
```js
cli.option('foo', 'What the foo?', '-f', '--foo', 'string');
cli.parse();

cli.params.foo === ['bar'];   // true  (actual)
```

I think having the `numValues` param default to `1` would be reasonable, and expected. I would also suggest splitting out the `numValues` param from the `required` param. If the options signature is changed as suggested in #7, it could look like this, then:

```js
cli.options('foo', {
  required: false,
  // numVals: 1  (default)
});
```

**Comment by four43** [comment]
I think splitting that out makes it more clear, especially when they are labeled as passed in JSON.

**Comment by four43** [comment]
Fixed via #16, now passing as options, but a length 1 returns a single value, length -1 (any) or length > 1 return arrays. See tests starting at: https://github.com/four43/admiral-cli/blob/master/test/OptionsTest.js#L290
